parser-typechecker/src/Unison/Codebase/Init.hs

```diff
@@ -42,6 +43,10 @@ data SpecifiedCodebase
   = CreateWhenMissing CodebasePath
   | DontCreateWhenMissing CodebasePath
 
+data CodebaseLockOption
```

Comment:
The semantics of when exactly to lock vs not isn't super clear to me, specifically with respect to Share,

We already get transactionality from Sqlite, so this lock must be just to protect us against our in-memory caches like the the root cache and Loop State values, but which operations in Enlil should lock vs not?

AFAICT we'd need to lock the codebase any time we intend to change the root branch, but my worry would be that it'd be pretty easy to add a codebase mutation to an existing share endpoint and forget to go back and add the lock on the top level. 🤔 

What locking strategy would you suggest we use on Share?
I think maybe no-locking is the right choice for most if not all endpoints and then we just try to make sure we're careful about using Transactions when needed, since share doesn't keep codebases open for long-periods like UCM does.

Reply:
I think share should not use the file locks (as we want concurrent reads on a codebase) and it should not use `getRootBranch` since it is very slow and not in Transaction.

I don't really know why we want a codebase-level root branch cache and the loop state cache for UCM. Do you know? Maybe we should delete the codebase-level root branch cache.

Reply:
Yeah I think we can delete the Codebase root branch cache probably, but do a quick audit for uses of `getRootBranch` across the unison repo first plz